### ISSUE TITLE
Is there a way to list exiting boxes and check if a certain box exists?

### ISSUE THREAD

**Issue by maxim-saplin** [hivedb/hive#375]
Didn't find api for:
1. Getting list of Hive boxes
2. Checking by box name if it has be created before (and/or is not empty)

Are there any options? Thanks

**Comment by TheMisir** [comment]
> 1. Getting list of existing/not empty Hive boxes

I think this feature have not yet implemented into hivedb.

> 2. Checking by box name if it has bern created before (and/or is not empty)

Did you tried using `.boxExists` method for checking exists boxes?

```dart
bool exists = await Hive.boxExists('boxName');
```

**Comment by maxim-saplin** [comment]
Thanks. It seems the way to check for existing boxes is to list *.hive files in app's direcotry (they are named after boxes' names).
P.S.: the use case for listing existing boxes is to keep track of potential grabage and clean it

**Comment by TheMisir** [comment]
> Thanks. It seems the way to check for existing boxes is to list *.hive files in app's direcotry (they are named after boxes' names).
> P.S.: the use case for listing existing boxes is to keep track of potential grabage and clean it

Yes you can list exists boxes by listing *.hive files. This was not implemented in hive because user may store hive files in any directory not just app's data dir.
